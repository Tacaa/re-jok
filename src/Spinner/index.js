//@flow
import * as React from 'react'
import {
  SpinBounce,
  SpinDefault
} from './style';


export type Props = {
  /** Set size of spinner component... default value is default**/
  size: 'small' | 'default' | 'medium' | 'large',
  /** Set type of spinner component... default value is default**/
  type: 'default' | 'bounce'
}

const defaultProps = {
  size: 'default',
  type: 'default'
}


class Spinner extends React.Component<Props> {

  static defaultProps = defaultProps;

  render(){
    const {
      type
    } = this.props;
    return(
      type === "default" ? <SpinDefault {...this.props}/> : <SpinBounce {...this.props}/>
    )
  }
}

export default Spinner
