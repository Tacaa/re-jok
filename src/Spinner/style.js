//@flow
import * as React from 'react';

import styled, { css, keyframes } from 'styled-components';

//Bounce Spinner
//This size  is different size another component
const returnSizeSpinner = props => {
  switch (props.size) {
    case 'small':
      return `
        width: 1.2rem;
        height: 1.2rem;
      `
    case 'default':
      return `
        width: 1.4rem;
        height: 1.4rem;
      `
    case 'medium' :
      return `
        width: 1.6rem;
        height: 1.6rem;
      `
    case 'large' :
      return `
       width: 2rem;
       height: 2rem;
      `
    default:
     return `
       width: 1.4rem;
       height: 1.4rem;
     `
  }
}


const skBouncedelay = keyframes`
0%, 80%, 100% {

 transform: scale(0);
} 40% {

 transform: scale(1.0);
}
`;


export const StyledSpinnerContainer = styled.div`

 ${'' /* width: 80px; */}
 ${'' /* text-align: center; */}
 & > div{
    ${props => returnSizeSpinner(props)};
    background-color: #333;
    border-radius: 100%;
    display: inline-block;
    -webkit-animation: ${skBouncedelay} 1.4s ease-in-out 0s infinite both;
            animation: ${skBouncedelay} 1.4s ease-in-out 0s infinite both;
 }
`

export const StyledSpinnerBounce1 = styled.div`

animation-delay: -0.32s !important;
`

export const StyledSpinnerBounce2 = styled.div`

animation-delay: -0.16s !important;
`
export const StyledSpinnerBounce3 = styled.div`

`

//Default Spinner
const spin = keyframes `
  to {transform: rotate(360deg);}
`;

export const StypedSpinnerDefault = styled.div `
display: inline-block;
width: ${props => props.theme.size[props.size].fontSize};
height: ${props => props.theme.size[props.size].fontSize};
min-width: ${props => props.theme.size[props.size].fontSize};
min-height: ${props => props.theme.size[props.size].fontSize};
position: relative;
color: inherit;

  &::before{
    content: '';
  box-sizing: border-box;
  position: absolute;
  top: 50%;
  left: 50%;
  width: ${props => props.theme.size[props.size].fontSize};
  height: ${props => props.theme.size[props.size].fontSize};
  margin-top: calc(${props => props.theme.size[props.size].fontSize} / -2);
  margin-left: calc(${props => props.theme.size[props.size].fontSize} / -2);
  border-radius: 50%;

  border-top: 2px solid;
  border-right: 2px solid transparent;
  animation: ${spin} .6s linear infinite;
  }
`


export const SpinBounce = props => (
  <React.Fragment>
    <StyledSpinnerContainer {...props}>
      <StyledSpinnerBounce1 />
      <StyledSpinnerBounce2 />
      <StyledSpinnerBounce3 />
    </StyledSpinnerContainer>
  </React.Fragment>
)

export const SpinDefault = props => (
  <React.Fragment>
    <StypedSpinnerDefault {...props} />
  </React.Fragment>
)
