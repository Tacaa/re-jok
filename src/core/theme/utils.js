import {
  defaultTheme
} from './index';


export const configTheme = (theme: Object) => {
  let jokTheme = {...defaultTheme, ...theme};
  return jokTheme;
}
