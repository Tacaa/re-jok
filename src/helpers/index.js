export const generalId = () => {
  return '_' + Math.random().toString(36).substr(2, 9);
}

//Default Function
export const noop = () => {}
