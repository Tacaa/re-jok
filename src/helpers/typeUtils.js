/** Check element has child or not**/
export const isChild = children => children === null
    || children === undefined
    || (Array.isArray(children) && children.length === 0)

/** Check undefined**/

export const isUndef = value => typeof value === 'undefined';
/** Check function or not**/

export const isFunction = value => !isUndef(value) && typeof value === 'function';

/** Check Object or not**/
export const isObject = value =>
      typeof value === 'object' && value !== null && !Array.isArray(value);

/** Check Array or not**/
export const isArray = value => Array.isArray(value);

/** Check Number or not**/
export const isNumber = value => Number.isInteger(value) && typeof value === "number"
