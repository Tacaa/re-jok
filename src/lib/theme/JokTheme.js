// @flow
import * as React from 'react';
import { ThemeProvider } from 'styled-components';
import { defaultTheme } from './index';

type Props = {
  /**Theme of project**/
  theme: Object,
  /**Children can be anything**/
  children: any,
}

const defaultProps = {
  theme: {}
}

class JokTheme extends React.Component<Props>{
  static defaultProps= defaultProps;

  checkDefaultTheme = ()=>{
    const {theme} = this.props;
    if(Object.keys(theme).length === 0 && theme.constructor === Object) return true;
    return false;
  }

  render(){
    const {
      children,
      theme
    } = this.props;

    let useDefaultTheme = this.checkDefaultTheme();

    return (
      <ThemeProvider theme={useDefaultTheme? defaultTheme : theme}>
        {children}
      </ThemeProvider>
    )
  }
}

export default JokTheme
