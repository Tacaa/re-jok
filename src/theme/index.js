// @flow
// export our theme to the provider
// This is globally available in styled-components when interpolating a function like so:
// ${(props) => props.theme}
// Or using import { withTheme } from 'styled-components';


export const defaultTheme = {
  size: {
    small: {
      fontSize: '1.2rem',
      padding: '0.5rem 0.9rem',
      "lineHeight": "1",
      "borderRadius": ".2rem",
      width: "2.4rem",
      height: "2.4rem"
    },
    default: {
      fontSize: '1.4rem',
      padding: '0.9rem 1.8rem',
      lineHeight: "1.5",
      borderRadius: ".4rem",
      width: "3.2rem",
      height: "3.2rem"
    },
    medium: {
      fontSize: '1.8rem',
      padding: '1.4rem 2.7rem',
      lineHeight: "1.5",
      borderRadius: ".6rem",
      width: "4rem",
      height: "4rem"

    },
    large: {
      fontSize: '2rem',
      padding: '1.8rem 3.6rem',
      lineHeight: "2",
      borderRadius: ".6rem",
      width: "4.8rem",
      height: "4.8rem"
    }
  },
  grid:{
    xs: 768,
    sm: 768,
    md: 992,
    lg: 1200,
    xl: 1920
  },
  text: {
    primary: "rgba(0, 0, 0, 0.85)",
    secondary: "rgba(0, 0, 0, 0.48)",
    hint: "rgba(0, 0, 0, 0.38)"
  },
  background: {
    paper: "#fff",
    default: "#fafafa"
  },
  typography: {
    title: {
      fontSize: "2rem",
      fontWeight: "600",
      lineHeight: "2.8rem"

    },
    subtitle: {
      fontSize: "1.6rem",
      fontWeight: "400",
      lineHeight: "2.4rem"
    }
  },
  primary: {
    main: "#1c8ffb",

  },
  secondary: {
    main: "#1b1c1d",
  },
  info: {
    main: "#1890ff"
  },
  error: {
    main: "#f5222d"
  },
  success: {
    main: "#52c41a"
  },
  warning: {
    main: "#faad14"
  },
  zIndex: {
    base: 1,
    background: 0,
    hidden: -1,
    card: 2,
    loading: 3,
    avatar: 3,
    form: 3,
    search: 3,
    badge: 100,
    modal: 1500,
  }
};

const changeTheme = (defaultTheme, theme)=>{
  let cloneTheme = {...defaultTheme};


  Object.keys(theme).forEach(function (key) {
    if ( 'object' === typeof theme[key] && !Array.isArray(theme[key]) ) {
    	changeTheme(cloneTheme[key], theme[key])
    }
    else if ( 'string' === typeof theme[key] || 'number' === typeof theme[key] ) {
      cloneTheme[key] = theme[key]
    }
  })
  return cloneTheme;
}

export const setTheme = (theme: Object) => {

   const newTheme = changeTheme(defaultTheme, theme);

   return newTheme;

}
