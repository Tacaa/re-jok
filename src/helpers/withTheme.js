import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme';
import {defaultTheme} from '../theme';
import { ThemeProvider } from 'styled-components';
export function renderWithTheme(component) {

  return renderer.create(
    <ThemeProvider theme={defaultTheme}>
      {component}
    </ThemeProvider>
  );
};

export const shallowWithTheme = (tree, theme) => {
  const context = shallow(<ThemeProvider theme={theme} />)
    .instance()
    .getChildContext()
  return shallow(tree, { context })
};

export const mountWithTheme = (tree, theme) => {
  const context = shallow(<ThemeProvider theme={theme} />)
    .instance()
    .getChildContext()

  return mount(tree, {
    context,
    childContextTypes: ThemeProvider.childContextTypes
  })
};
