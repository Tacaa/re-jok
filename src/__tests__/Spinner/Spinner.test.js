import React from 'react';
import Spinner from '../../Spinner';


import {defaultTheme} from '../../theme';

import {shallowWithTheme, mountWithTheme, renderWithTheme} from '../../helpers/withTheme';
import 'jest-enzyme';
import 'jest-styled-components';

describe('<Spinner />', () => {
  it('should render default', () => {
    const tree = renderWithTheme(
      <Spinner />

    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render with size', () => {
    const tree = renderWithTheme(
      <Spinner size="medium" />
    ).toJSON();
    expect(tree).toMatchSnapshot();


  });

  it('should render type bounce', () => {
    const tree = renderWithTheme(
      <Spinner type="bounce"/>


    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render type bounce with size', () => {
    const tree = renderWithTheme(
      <Spinner type="bounce" size="medium"/>


    ).toJSON();
    expect(tree).toMatchSnapshot();


  });





})
