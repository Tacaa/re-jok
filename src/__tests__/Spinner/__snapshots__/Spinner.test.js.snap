// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Spinner /> should render default 1`] = `
.c0 {
  display: inline-block;
  width: 1.4rem;
  height: 1.4rem;
  min-width: 1.4rem;
  min-height: 1.4rem;
  position: relative;
  color: rgba(0,0,0,.85);
}

.c0::before {
  content: '';
  box-sizing: border-box;
  position: absolute;
  top: 50%;
  left: 50%;
  width: 1.4rem;
  height: 1.4rem;
  margin-top: calc(1.4rem / -2);
  margin-left: calc(1.4rem / -2);
  border-radius: 50%;
  border-top: 2px solid;
  border-right: 2px solid transparent;
  -webkit-animation: ibwhYF .6s linear infinite;
  animation: ibwhYF .6s linear infinite;
}

<div
  className="c0"
  size="default"
  type="default"
/>
`;

exports[`<Spinner /> should render type bounce 1`] = `
.c0 > div {
  width: 1.4rem;
  height: 1.4rem;
  background-color: rgba(0,0,0,.85);
  border-radius: 100%;
  display: inline-block;
  -webkit-animation: ghjfcd 1.4s ease-in-out 0s infinite both;
  -webkit-animation: ghjfcd 1.4s ease-in-out 0s infinite both;
  animation: ghjfcd 1.4s ease-in-out 0s infinite both;
}

.c1 {
  -webkit-animation-delay: -0.32s !important;
  animation-delay: -0.32s !important;
}

.c2 {
  -webkit-animation-delay: -0.16s !important;
  animation-delay: -0.16s !important;
}

<div
  className="c0"
  size="default"
  type="bounce"
>
  <div
    className="c1"
  />
  <div
    className="c2"
  />
  <div
    className=""
  />
</div>
`;

exports[`<Spinner /> should render type bounce with size 1`] = `
.c0 > div {
  width: 1.6rem;
  height: 1.6rem;
  background-color: rgba(0,0,0,.85);
  border-radius: 100%;
  display: inline-block;
  -webkit-animation: ghjfcd 1.4s ease-in-out 0s infinite both;
  -webkit-animation: ghjfcd 1.4s ease-in-out 0s infinite both;
  animation: ghjfcd 1.4s ease-in-out 0s infinite both;
}

.c1 {
  -webkit-animation-delay: -0.32s !important;
  animation-delay: -0.32s !important;
}

.c2 {
  -webkit-animation-delay: -0.16s !important;
  animation-delay: -0.16s !important;
}

<div
  className="c0"
  size="medium"
  type="bounce"
>
  <div
    className="c1"
  />
  <div
    className="c2"
  />
  <div
    className=""
  />
</div>
`;

exports[`<Spinner /> should render with size 1`] = `
.c0 {
  display: inline-block;
  width: 1.8rem;
  height: 1.8rem;
  min-width: 1.8rem;
  min-height: 1.8rem;
  position: relative;
  color: rgba(0,0,0,.85);
}

.c0::before {
  content: '';
  box-sizing: border-box;
  position: absolute;
  top: 50%;
  left: 50%;
  width: 1.8rem;
  height: 1.8rem;
  margin-top: calc(1.8rem / -2);
  margin-left: calc(1.8rem / -2);
  border-radius: 50%;
  border-top: 2px solid;
  border-right: 2px solid transparent;
  -webkit-animation: ibwhYF .6s linear infinite;
  animation: ibwhYF .6s linear infinite;
}

<div
  className="c0"
  size="medium"
  type="default"
/>
`;
